src/wp-includes/blocks.php

```diff
@@ -402,37 +418,24 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 	 * instead of reading a JSON file per-block, and then decoding from JSON to PHP.
 	 * Using a static variable ensures that the metadata is only read once per request.
 	 */
-	static $core_blocks_meta;
-	if ( ! $core_blocks_meta ) {
-		$core_blocks_meta = require ABSPATH . WPINC . '/blocks/blocks-json.php';
-	}
 
 	$metadata_file = ( ! str_ends_with( $file_or_folder, 'block.json' ) ) ?
 		trailingslashit( $file_or_folder ) . 'block.json' :
 		$file_or_folder;
 
-	$is_core_block = str_starts_with( $file_or_folder, ABSPATH . WPINC );
-	// If the block is not a core block, the metadata file must exist.
-	$metadata_file_exists = $is_core_block || file_exists( $metadata_file );
-	if ( ! $metadata_file_exists && empty( $args['name'] ) ) {
-		return false;
-	}
-
-	// Try to get metadata from the static cache for core blocks.
-	$metadata = array();
-	if ( $is_core_block ) {
-		$core_block_name = str_replace( ABSPATH . WPINC . '/blocks/', '', $file_or_folder );
-		if ( ! empty( $core_blocks_meta[ $core_block_name ] ) ) {
-			$metadata = $core_blocks_meta[ $core_block_name ];
-		}
-	}
-
-	// If metadata is not found in the static cache, read it from the file.
-	if ( $metadata_file_exists && empty( $metadata ) ) {
-		$metadata = wp_json_file_decode( $metadata_file, array( 'associative' => true ) );
-	}
-
-	if ( ! is_array( $metadata ) || ( empty( $metadata['name'] ) && empty( $args['name'] ) ) ) {
+	$metadata             = null;
+	$metadata_file_exists = false;
+	$registry_metadata    = WP_Block_Metadata_Registry::get_metadata( $file_or_folder );
+
+	if ( $registry_metadata ) {
+		$metadata             = $registry_metadata;
+		$metadata_file_exists = file_exists( $metadata_file );
+	} else if ( file_exists( $metadata_file ) ) {
+		$metadata             = wp_json_file_decode( $metadata_file, array( 'associative' => true ) );
+		$metadata_file_exists = true;
+	} else if ( ! empty( $args['name'] ) ) {
```

Comment:
```suggestion
	} elseif ( ! empty( $args['name'] ) ) {
```

Per https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#use-elseif-not-else-if, `Use elseif, not else if`